treeherder/model/derived/jobs.py

```diff
@@ -1702,14 +1704,20 @@ def store_result_set_data(self, result_sets):
         # TODO: Confirm whether we need to do a lookup in this loop in the
         #   memcache to reduce query overhead
         for result in result_sets:
-
+            top_revision = result['revisions'][-1]
```

Comment:
good call on this.  I'll add a test, and you're right about that change being cleaner.  Doing that now.